src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -161,6 +171,11 @@ public void RegisterOperationAction(Action<OperationAnalysisContext> action, par
             this.RegisterOperationAction(action, operationKinds.AsImmutableOrEmpty());
         }
 
+        internal void RegisterOperationActionParamsArrayInternal(Action<OperationAnalysisContext> action, params OperationKind[] operationKinds)
```

Comment:
Why include "ParamsArray" or "ImmutableArray" in the name? Can we keep it consistent with existing methods and just call them "RegisterOperationActionInternal"?